Fix: App icon in launch (aka splash) screen should not move (up) abruptly #1897

hadwanihardik opened the pull request:
Fixes #1897

hboon commented on AlphaWallet/Extensions/UIDevice.swift:
```diff
@@ -0,0 +1,16 @@
+import UIKit
+
+extension UIDevice {
+    var hasNotch: Bool {
```
Can you fix it without add `hasNotch`? This is very brittle.

hboon commented on AlphaWallet/Protection/Coordinators/ProtectionCoordinator.swift:
```diff
@@ -21,7 +21,7 @@ class ProtectionCoordinator: Coordinator {
 
 	func didFinishLaunchingWithOptions() {
-		splashCoordinator.start()
+		//splashCoordinator.start()
```
Commented out code should generally not be checked in, they should be deleted.

What is the rationale for removing this though?

hadwanihardik replied:
Actually this part to code is adding splash view on window even before app is loaded so its not getting proper frame coordinates. because of that when we open app after creating wallet it shows black screen with logo with layout issue.
`splashCoordinator.start()`
As its already called on `applicationDidBecomeActive` we can remove this line of code in 
`didFinishLaunchingWithOptions`.

hboon commented on AlphaWallet/Protection/Coordinators/SplashCoordinator.swift:
```diff
@@ -9,13 +9,16 @@ class SplashCoordinator: Coordinator {
 
     init(window: UIWindow) {
         self.window = window
+        window.backgroundColor =  .white;
```
What is this for?

hadwanihardik replied:
This was making background color of splash view controller white as on app launch when its loaded on window object it was showing black  screen with logo after wallet creation

hboon commented on AlphaWallet/Protection/Coordinators/SplashCoordinator.swift:
```diff
@@ -9,13 +9,16 @@ class SplashCoordinator: Coordinator {
     func start() {
         window.rootViewController = SplashViewController()
         window.isHidden = false
+        window.rootViewController = SplashViewController()
```
Why is it being created and assigned twice?

hadwanihardik replied:
it was by mistake removing it. one part of it was commented.

hboon commented on AlphaWallet/Protection/ViewControllers/SplashViewController.swift:
```diff
@@ -8,14 +8,29 @@ class SplashViewController: UIViewController {
+        splashView.frame =  UIScreen.main.bounds
+       view.addSubview(splashView)
+       if UIDevice.current.hasNotch {
+        if #available(iOS 11, *) {
```
* We should always follow the existing indentation style in the file (or if that file is really bad, fix the file).
* The app doesn't support < iOS 11, so the check is not needed

hadwanihardik replied:
ok sure I am making note of this.

hboon commented on AlphaWallet/Protection/ViewControllers/SplashViewController.swift:
```diff
@@ -8,14 +8,29 @@ class SplashViewController: UIViewController {
+        } else {
+           let standardSpacing: CGFloat = 8.0
+           NSLayoutConstraint.activate([
+           splashView.topAnchor.constraint(equalTo: view.bottomAnchor, constant: standardSpacing),
```
Indentation, again

hadwanihardik replied:
ok

hboon commented on AlphaWallet/Protection/ViewControllers/SplashViewController.swift:
```diff
@@ -8,14 +8,29 @@ class SplashViewController: UIViewController {
+        splashView.layoutSubviews();
+
+    }
+    override func viewDidLayoutSubviews() {
```
If it's empty, we should remove it

hadwanihardik replied:
ok

hboon commented on AlphaWallet/Protection/Views/SplashView.swift:
```diff
@@ -3,16 +3,31 @@
+    }
+    override func layoutSubviews() {
+        if UIDevice.current.hasNotch {
+                   let guide = safeAreaLayoutGuide
```
Indentation. Ca you fix the indentation for the entire commit?

hadwanihardik replied:
ok

hboon commented on Podfile.lock:
```diff
@@ -252,4 +252,4 @@ SPEC CHECKSUMS:
 PODFILE CHECKSUM: 4d513ea9af5a9550a53c8b47493983d196f33d13
 
-COCOAPODS: 1.8.4
+COCOAPODS: 1.9.1
```
Generally, we try not to include any unnecessary changes in a PR

hadwanihardik replied:
ok

hadwanihardik commented:
Made Changes as per suggestion by boon.

1. Made indentation proper for all files in commit.
2. Removed unnecessary commented code and spaces.
3. Removed condition for check ios version lower than ios 11.
4. Removed code for checking notch display, and removed extension file related to same.
5. Removed blank methods.

hboon commented:
Thanks. Would you help to rebase master and squash your changes into a single commit, and I’ll check again?

hadwanihardik commented:
> Thanks. Would you help to rebase master and squash your changes into a single commit, and I’ll check again?

Hey I did combined changes in single commit.

hboon commented:
It seems like you might have accidentally modified the 2nd commit in this PR. Can you compare to master? Your PR branch should basically be the tip of master + 1 commit

hadwanihardik commented:
> It seems like you might have accidentally modified the 2nd commit in this PR. Can you compare to master? Your PR branch should basically be the tip of master + 1 commit

Hey I did it by mistake, first commit is not for this PR. How can i remove it from PR?

hboon commented:
Rebase

hadwanihardik commented:
> Rebase

I did not understood. I tried to rebase to my commit but it did not worked

hboon commented:
Could you have accidentally squashed your commit with the commit at the tip of the master?

hadwanihardik commented:
> Could you have accidentally squashed your commit with the commit at the tip of the master?

I don't have idea. if u can guide me?

hadwanihardik commented:
> > Could you have accidentally squashed your commit with the commit at the tip of the master?
> 
> I don't have idea. if u can guide me?

may I generate a new PR with new branch?

hboon commented:
> may I generate a new PR with new branch?

Ok

hboon commented:
Superseded by #1966
